russstation/code/game/objects/items/miscellaneous.dm

```diff
@@ -0,0 +1,172 @@
+		else
+			to_chat(user, "<span class='notice'>You add a [I.name] to the bottom of the [src.name].<span>")
+			qdel(I)
+			qdel(src)
+			new /obj/item/clothing/suit/caution/incomplete(L, 1)
+
+		
+/obj/item/clothing/suit/caution/incomplete
+	name = "Incomplete wet floor sign"
+	slot_flags = null
+
+/obj/item/clothing/suit/caution/incomplete/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The [src.name] has a floor buffer underneath. You could <i>attach a sensor</i> to it, or <i>undo the screws</i> to remove it.</span>"
+
+/obj/item/clothing/suit/caution/incomplete/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	var/atom/L = drop_location()
+
+	if(istype(I, /obj/item/screwdriver))
+		to_chat(user, "<span class='notice'>You detatch the floor buffer from the [src.name].<span>")
+		qdel(src)
+		new /obj/item/clothing/suit/caution(L, 1)
+		new /obj/item/janiupgrade(L, 1)
+
+	if(istype(I, /obj/item/assembly/prox_sensor))
+		to_chat(user, "<span class='notice'>You add a [I.name] to the floor buffer on the [src.name].<span>")
+		qdel(I)
+		qdel(src)
+		var/obj/item/S = /obj/item/clothing/suit/caution/slippery
+		if(prob(1)) //1% chance for a name
+			S.name = pick("Ms. Lippy", "Mr. Walky", "Monitor Hallsky")
+
+		new S(L, 1)
+
+//old signs (only found in maint spawners)
+/obj/item/caution/attackby(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I, /obj/item/janiupgrade))
+		to_chat(user, "<span class='warning'>The [src.name] is too antiquated to fit a [I.name], try a newer sign.<span>")
+
+
+//end of construction code
+
+//Wetmore Slippery Sign
+//Constructable by janitors with upgrades from RND and sensors from cargo/robotics
+//Slips people who don't heed the warning
+
+//Wetmore Slippery Sign (traitor item)
+//purchasable from traitor uplink by janitors (7tc for 4)
+//in addition to slipping, also animates and kicks people around
+
+/obj/item/clothing/suit/caution/slippery
+	slot_flags = null //cannot be worn- or else it breaks stuff
+
+	var/willSlip = 0 //0 - disabled || 1 - enabled 
+	var/lastSlip = 0 //last time the sign slipped someone (time)
+	var/slipCooldown = 50 //cooldown (~5 seconds default)
+	var/clowningAround = 0 //0 - untainted || 1 - clown got it
+	var/evilSign = 0 //is the sign a traitor?
+	var/mob/living/carbon/boss = null //used so animated signs don't attack the janitor
+
+/obj/item/clothing/suit/caution/slippery/syndicate
+	evilSign = 1 //signs purchased from the uplink
+
+/obj/item/clothing/suit/caution/slippery/examine(mob/user)
+	. = ..()
+	if(isobserver(user) || user.mind.assigned_role == "Janitor") //janitors and ghosts can see that it's a slippery sign
+		. += "<span class='notice'>This [src.name] is outfitted with an experimental sprayer.</span>"
+		if(clowningAround || evilSign)
+			. += "<span class='warning'>It's been tampered with.</span>"
+	else if(HAS_TRAIT(user, TRAIT_CLUMSY)) //clowns know lube when they see it
```

Comment:
Clumsy can be gotten by non-clowns. Is this intended? Or would checking against `user.mind.assigned_role` be closer to your intent?

Reply:
this is intentional